internal/install/execution/install_status.go

```diff
@@ -175,6 +175,17 @@ func (s *InstallStatus) RecipeSkipped(event RecipeStatusEvent) {
 	}
 }
 
+func (s *InstallStatus) InstallError(err error) {
```

Comment:
It looks to me like if `err` is nil, then the status will still contain `unknown error` here.  I'm wondering if we could smooth out the error handling here.  Perhaps `InstallComplete()` could take the error here rather than a separate method for it.

Reply:
@zlesnr 
I'm not quite understand go enough here (channel wiring), can you tell me what would cause the caller to get into this flow with a nil error?
I know the tests did that, hence the nil check, is there a valid usage when installing where we would error without an actual error object?

Reply:
`InstallComplete` method is meant to handle the success case as well as the error case.  In the success case, `err` will be nil.  I would get rid of this `InstallError` method, and instead do this check in `InstallComplete`:

```
if err != nil, set the nerdstorage error
```

Reply:
Ok will do, thanks @ctrombley